src/entity.ts

```diff
@@ -163,6 +195,14 @@ export class Entity {
 		else
 			properties = property;
 
+		if (!this._context) {
+			const context = new InitializationContext(true);
+			context.execute(() => {
+				this.updateWithContext(context, properties);
+			});
+			return context.ready;
+		}
```

Comment:
This facilitates updates being asynchronous by default.

Reply:
Is it async by default or always async? This essentially re-enters the `updateWithContext` with a context if one isn't already present, which it looks to me like that would typically be the case when calling `update` directly, whereas when called via `updateWithContext` it would obviously already be defined and continue to the property setting below. Does this mean that `update` calls will always be async, or is there some condition where `_context` would _not_ be nulled out in `updateWithContext`?

Reply:
Yeah, that was poor wording on my part. The function always returns a `Promise`, so in that sense it is async always. I think what I was trying to get at is the fact that, in the event the user has not added any asynchrony to their model, you can treat `update()` as a synchronous operation.

```ts
entity.update({ Field1: 1, Field2: 2 });

console.log(entity.Field1, entity.Field2); // 1, 2
```

Reply:
Good deal 👍